src/main/java/io/stargate/sgv2/jsonapi/exception/playing/EmbeddingProviderException.java

```diff
@@ -0,0 +1,28 @@
+package io.stargate.sgv2.jsonapi.exception.playing;
+
+public class EmbeddingProviderException extends ServerException {
+  public EmbeddingProviderException(ErrorInstance errorInstance) {
+    super(errorInstance);
+  }
+
+  public enum Code implements ErrorCode<EmbeddingProviderException> {
```

Comment:
Is the idea it's ok to have same "code" (like CLIENT_ERROR) for multiple scopes? I guess that is the case.

One concern is that to know which problem you have, it is no longer possible to group failures solely by Code, as that will conflate Codes with different Scopes.

Reply:
will add a fully qualified code to the API exception, we can then use it later for metrics etc